### PR TITLE
Added get user list feature

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary

Added get user list feature

## Trello Card

https://trello.com/c/FDgXzMBH/4-lista-de-usuarios

### CODE REVIEW

File: app/helpers/utils.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ const jwt = require('jsonwebtoken');
 const errors = require('../errors');
 const config = require('../../config');
 
+const DEFAULT_PAGE = 0;
+const DEFAULT_LIMIT = 10;
```
Comment (arinaldi118) [inline app/helpers/utils.js]: These constants must be in a constant file

Comment (arinaldi118) [reply app/helpers/utils.js]: And the first page must be 1

File: app/helpers/utils.js [hunk 2]
```diff
@@ -21,8 +24,18 @@ const createToken = payload => {
   };
 };
 
+const paginate = (page = DEFAULT_PAGE, limit = DEFAULT_LIMIT) => {
+  const offset = page * limit;
```
Comment (arinaldi118) [inline app/helpers/utils.js]: (page -1) * limit

File: app/middlewares/authenticator.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const jwt = require('jsonwebtoken');
+const config = require('../../config');
+const errors = require('../errors');
+const logger = require('../logger');
+
+module.exports = (req, res, next) => {
+  if (!req.headers.authorization) {
+    return next(errors.unauthenticatedUserError('You need to sign in to access this resource'));
+  }
+  const token = req.headers.authorization.slice(7);
```
Comment (arinaldi118) [inline app/middlewares/authenticator.js]: `slice(7)`? why?

File: app/schemas/user.js [hunk 4]
```diff
@@ -41,7 +41,31 @@ const signUp = {
   }
 };
 
+const listAllUsers = {
+  page: {
+    in: ['query'],
+    exists: {
```
Comment (arinaldi118) [inline app/schemas/user.js]: :thinking: This must be optional. If the request doesn't have a page you must return page 1.

File: app/schemas/user.js [hunk 5]
```diff
@@ -41,7 +41,31 @@ const signUp = {
...
+      errorMessage: 'Missing page number param'
+    },
+    isInt: {
+      options: { min: 1 },
+      errorMessage: 'page param must be an integer greater than 1'
+    }
+  },
+  limit: {
```
Comment (arinaldi118) [inline app/schemas/user.js]: Idem page.

File: app/schemas/user.js (hunk 4 above)
Comment (tagonzalez) [reply app/schemas/user.js]: Solved!

File: app/schemas/user.js (hunk 5 above)
Comment (tagonzalez) [reply app/schemas/user.js]: Solved!

**Pull request merged**